script.js

```diff
@@ -1,4 +1,696 @@
+var mode = "playersetup";
+var playerBlackjack = false;
+var compBlackjack = false;
+var compHandText = "";
+var playerHandText = "";
+var playerHand = [];
+var compHand = [];
+var playerCard1 = {};
+var playerCard2 = {};
+var computerCard1 = {};
+var computerCard2 = {};
+var noOfPlayers = 0;
+var allPlayers = [];
+var playerTotalValue = 0;
+var compTotalValue = 0;
+var playerStandText = "";
+var username = "";
+var dealIndex = 0;
+var cardDisplayCounter = 0;
+var playerIndex = 0;
+var splitPlayer = {};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (mode == "playersetup" || mode == "inputname" || mode == "betmode") {
+    compHand = [];
+    return playerArraySetup(input);
+  }
+  if (mode == "dealmode" || mode == "compdeal") {
+    shuffledDeck = shuffleCards(deck);
+    var myOutputValue = dealCards(shuffledDeck);
+    return myOutputValue;
+  }
+  if (
+    mode == "playerturn" ||
+    mode == "playerDecisionTurn" ||
+    mode == "playerEndTurn" ||
+    mode == "splitDecision"
+  ) {
+    var playerTurnText = playerTurnFunc(input);
+    return playerTurnText;
+  }
+  if (mode == "compturn") {
+    var myOutputValue = compTurnFunc(compHand);
+  }
```

Comment:
great job simplifying the code in the main function. An alternative you might consider since you have quite a few conditions is to use a switch statement. https://www.w3schools.com/js/js_switch.asp